### DISCUSSION TITLE
[Question] Issues Packing obfuscated scripts manually. Some python scripts get packed not obfuscated.

### DISCUSSION THREAD

**Discussion by Warfront1** [dashingsoft/pyarmor#1415] [General]
I have been trying to follow the documentation written at the following link.
https://pyarmor.readthedocs.io/en/latest/topic/repack.html?highlight=--pack#packing-obfuscated-scripts-manually

I have automated all of the manual steps the process has suggested into a build script (`build.sh`).
https://github.com/Warfront1/pyarmor-macos-multi-architecture-test-public/tree/feat/test-manual-pack

The scripts are all properly obfuscated, and are able to be ran prior to packing.
https://github.com/Warfront1/pyarmor-macos-multi-architecture-test-public/actions/runs/5791515509/job/15696447693#step:4:54

However once I attempt to pack them using the provided patch and PyInstaller we find out that PyInstaller packs non-obfuscated scripts.
https://github.com/Warfront1/pyarmor-macos-multi-architecture-test-public/actions/runs/5791515509/job/15696447693#step:4:120

It would appear that there is something wrong with the pyarmor `.spec` patch not working properly?
I've added various debugging print statements to the patch code to ease with debugging.
It looks like it is properly replacing the `a.pure` list, but perhaps the pyarmor patch code is missing something else?

**Comment by jondy** [comment]
All of them are open source, could you find it why?
